src/main/scala/piconot/internal/ourDSL.scala

```diff
@@ -0,0 +1,132 @@
+ * So, we must cast both types of tuples to ShortRule.
+ */
+
+class ourDSL extends JFXApp {
+
+  // Initializes the picobot and runs graphics display
+  def picobot(mazeName: String)(rules: Seq[Rule]*) = {
+    val rulesList = rules.flatten.toList
+    val maze = Maze("resources" + File.separator + mazeName + ".txt")
+    object EmptyBot extends Picobot(maze,rulesList)
+      with TextDisplay with GUIDisplay
+    stage = EmptyBot.mainStage
+    EmptyBot.run()
+  }
+
+  // Allows users to group by state a series of rules
+  def state(stateName: State)(moves: ShortRule*): Seq[Rule] = {
+    moves.map(r => Rule(stateName, r.s, r.m, r.state))
+  }
+
+  // Natural language method to set move direction
+  def move(dir: MoveDirection): MoveDirection = dir
+  
+  // Natural language method to set end state
+  def to(end: String): State = end
+  
+  // Passes variatic MoveDirection and uses pattern matching to check using relative description
+  // which direction is blocked
+  def blocked(walls: MoveDirection*): OurSurroundings = {
+    var northWall, eastWall, westWall, southWall: RelativeDescription = Anything
+    walls.foreach { dir =>
+      dir match {
+        case North => northWall = Blocked
+        case East  => eastWall = Blocked
+        case West  => westWall = Blocked
+        case South => southWall = Blocked
+      }
+    }
+    OurSurroundings(northWall, eastWall, westWall, southWall)
+  }
+
+  // Similar to blocked, but checks if the relative description is open
+  def open(walls: MoveDirection*): OurSurroundings = {
+    var northWall, eastWall, westWall, southWall: RelativeDescription = Anything
+    walls.foreach { dir =>
+      dir match {
+        case North => northWall = Open
+        case East  => eastWall = Open
+        case West  => westWall = Open
+        case South => southWall = Open
+      }
+    }
+    OurSurroundings(northWall, eastWall, westWall, southWall)
+  }
+
+  // Casts a tuple to ShortRule
+  implicit def tupletoShortRule(tuple: (Surroundings, MoveDirection, State)): ShortRule = {
+    ShortRule(s = tuple._1, m = tuple._2, state = tuple._3)
+  }
+
+  // ShortRule expects a Surroundings, so we explicitly construct one from OurSurroundings.
+  // Making an alternate constructor for ShortRule that takes an OurSurroundings didn't work for some reason (even though I tried using the 'new' keyword).
+  implicit def tupletoShortRule2(tuple: (OurSurroundings, MoveDirection, State)): ShortRule = {
+    ShortRule(s = Surroundings(tuple._1.north, tuple._1.east, tuple._1.west, tuple._1.south), m = tuple._2, state = tuple._3)
+  }
+  
+  // Implicit cast from string to State
+  implicit def stringToState(state: String): State = State(state)
+  
+  // Default vals for natural language replacement of cardinal directions
+  val left, west, w = West
+  val up, north, n = North
+  val right, east, e = East
+  val down, south, s = South
+  val stay, stop = StayHere
```

Comment:
I like that you defined multiple ways to reference the directions. This certainly gives the users a lot of power.